storage.js

```diff
@@ -0,0 +1,29 @@
+const nameId = document.getElementById('name');
+const emailId = document.getElementById('email');
+const commentId = document.getElementById('comment-box');
+const body = document.querySelector('body');
+
+const formDetails = JSON.parse(localStorage.getItem('contactInfo'));
+if (formDetails) {
+  nameId.value = formDetails.name;
+  emailId.value = formDetails.email;
+  commentId.value = formDetails.comment;
+}
+
+body.addEventListener('load', (event) => {
+  event.preventDefault();
+  const name = nameId.value;
+  const email = emailId.value;
+  const comment = commentId.value;
+
+  if (!name || !email || !comment) {
+    return;
+  }
+  // Store data on array//
```

Comment:
Unfortunately, your contact form still not preserving data in localStorage API. Because you need an event listener listens the change on the input area and triggered. A code similar to this `nameInput.addEventListener('input', populateStorage);
` You can check this project to understand how it needs to be implemented: [project](https://huseyinbiyik.github.io).